src/firmware_new/tools/debug_firmware_on_arm_board.sh

```diff
@@ -0,0 +1,143 @@
+#!/bin/bash
+#
+# This script allows the user to automatically flash and debug a given binary to a
+# ARM MCU over USB and debug it, all via OpenOCD
+#
+
+# See here for why we need this:
+# https://github.com/bazelbuild/bazel/blob/8fa6b3fe71f91aac73c222d8082e75c69d814fa7/tools/bash/runfiles/runfiles.bash#L38-L59
+# --- begin runfiles.bash initialization ---
+# Copy-pasted from Bazel's Bash runfiles library (tools/bash/runfiles/runfiles.bash).
+set -euo pipefail
+if [[ ! -d "${RUNFILES_DIR:-/dev/null}" && ! -f "${RUNFILES_MANIFEST_FILE:-/dev/null}" ]]; then
+  if [[ -f "$0.runfiles_manifest" ]]; then
+    export RUNFILES_MANIFEST_FILE="$0.runfiles_manifest"
+  elif [[ -f "$0.runfiles/MANIFEST" ]]; then
+    export RUNFILES_MANIFEST_FILE="$0.runfiles/MANIFEST"
+  elif [[ -f "$0.runfiles/bazel_tools/tools/bash/runfiles/runfiles.bash" ]]; then
+    export RUNFILES_DIR="$0.runfiles"
+  fi
+fi
+if [[ -f "${RUNFILES_DIR:-/dev/null}/bazel_tools/tools/bash/runfiles/runfiles.bash" ]]; then
+  source "${RUNFILES_DIR}/bazel_tools/tools/bash/runfiles/runfiles.bash"
+elif [[ -f "${RUNFILES_MANIFEST_FILE:-/dev/null}" ]]; then
+  source "$(grep -m1 "^bazel_tools/tools/bash/runfiles/runfiles.bash " \
+    "$RUNFILES_MANIFEST_FILE" | cut -d ' ' -f 2-)"
+else
+  echo >&2 "ERROR: cannot find @bazel_tools//tools/bash/runfiles:runfiles.bash"
+  exit 1
+fi
+set +e
+# --- end runfiles.bash initialization ---
+
+# The directory this script is in
+CURR_DIR=$(dirname -- "$(readlink -f -- "$BASH_SOURCE")")
+
+# The name of this script
+THIS_SCRIPT_FILENAME=$(basename "$0")
+
+# The root directory of this project (ie. the location of the WORKSPACE file). This is a
+# bit of a hack, but it works.
+WORKSPACE_DIR="$CURR_DIR/../../"
+
+# Enable bazel runtime debugging
+#RUNFILES_LIB_DEBUG=1
```

Comment:
Are we going to use this in the future? Is it worth keeping around?

Reply:
Yeah, if stuff goes wrong, we print an error suggesting to the user to uncomment it  for debugging